### PR TITLE
Use Cint to match C++'s int type, not Julia's Int.

### PR DESCRIPTION
**Pull request by rcurtin**
I mistakenly thought that Julia's `Int` type always matched C/C++'s `int` type; this is incorrect---Julia's `Int` type instead matches `size_t`.  Therefore, we have to use the `Cint` type in Julia.  This PR fixes that and all associated handling, and should fix the currently-failing Julia build.

### CODE REVIEW

**Comment by Yashwants19** [comment]
Hi @rcurtin I want to ask that can we can Cdouble in place of Float64.

**Comment by rcurtin** [comment]
@Yashwants19 correct me if I'm wrong, but it seems to me on https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/#man-bits-types-1 like `Float64` and `Cdouble` are entirely equivalent types on every system.  (At least according to that table...)

**Comment by Yashwants19** [comment]
[https://github.com/mlpack/mlpack/blob/f482383a8bb0f915829510879afa604cc514cabd/src/mlpack/bindings/julia/mlpack/cli.jl.in#L213](url)
I think so here we are passing size_t from julia but accept const int from c++ correct if i am wrong.

**Comment by Yashwants19** [comment]
> @Yashwants19 correct me if I'm wrong, but it seems to me on https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/#man-bits-types-1 like `Float64` and `Cdouble` are entirely equivalent types on every system. (At least according to that table...)

Sorry I think so I had overlooked this part. :)

File: .ci/windows-steps.yaml [hunk 1]
```diff
@@ -66,6 +66,7 @@ steps:
     -DARMADILLO_LIBRARY="..\armadillo-8.400.0\Release\armadillo.lib" `
     -DBOOST_INCLUDEDIR=$(Agent.ToolsDirectory)\boost.1.60.0.0\lib\native\include `
     -DBOOST_LIBRARYDIR=$(Agent.ToolsDirectory)\boost_libs `
+    -DBUILD_JULIA_BINDINGS=OFF `
```
Comment (zoq) [inline .ci/windows-steps.yaml]: Hm, that is strange looks like we are still trying to run the julia bindings tests.

Comment (rcurtin) [reply .ci/windows-steps.yaml]: Yeah, the debugging exposed a bug in the CMake configuration. :smile: I think 3f07bcf fixes it.

Comment (zoq) [reply .ci/windows-steps.yaml]: Nice, I guess with https://github.com/mlpack/mlpack/pull/2278 we should see more green builds.

Comment (rcurtin) [reply .ci/windows-steps.yaml]: Here's hoping!  It's been a really bad week for our build systems and I think a good part of that is my fault. :smile:

**Comment by rcurtin** [comment]
> [https://github.com/mlpack/mlpack/blob/f482383a8bb0f915829510879afa604cc514cabd/src/mlpack/bindings/julia/mlpack/cli.jl.in#L213](url)
> I think so here we are passing size_t from julia but accept const int from c++ correct if i am wrong.

Nice catch!  Thank you.  Fixed in 4bb869a.

**Comment by rcurtin** [comment]
I will wait until Azure builds this time :smile:

**Comment by zoq** [comment]
Hm, let's see if we can bring the azure build back.

**Comment by zoq** [comment]
Manually started the build: https://dev.azure.com/mlpack/mlpack/_build/results?buildId=998&view=results not sure why the last commit hasn't triggered the build.

**Comment by favre49** [comment]
Looks like all the builds passed (besides MacOS) :). I'll go ahead and merge it

**Pull request merged**